python/cudf/cudf/tests/test_replace.py

```diff
@@ -183,7 +182,7 @@ def test_series_fillna_numerical(
     if isinstance(fill_value, pd.Series):
         fill_value_cudf = cudf.from_pandas(fill_value)
     else:
-        fill_value_cudf = fill_value
+        fill_value_cudf = copy.deepcopy(fill_value)
```

Comment:
Just curious, why do we need to do this?

Reply:
When running this test with more parallelism in pytest-xdist we were running into issues of incorrect `fill_value_cudf` value in some tests.

Reply:
That sounds bad.. Any idea why? Is that a pytest-xdist bug or something?

Reply:
I think we're modifying the `df` that's passed in via pytest params inplace and it's likely that pytest initializes once and reuses the initialized values.

Reply:
Hmm, why would deepcopying the other parameter `fill_value` help with that though?

Reply:
I'm not clear, apologies was looking at the wrong deep copy. Will let @galipremsagar answer 😄

Reply:
we don't need a deepcopy here, apologies for the confusion caused. I did some digging and increased the cores count(`n`) to 24 to see what happens. So what was happening in `test_fillna_dataframe` started to show up in `test_series_fillna_numerical` where we were performing inplace operation and pytest or pytest-xdist was just reusing the same variable.

This surfaced another test failure in `test_series_fillna_numerical` where it shouldn't be passing in a serial/parallel run at all. Reason:
We have two float series and we are performing a type-cast to `data_dtype`(let says it is `uint64`) it would error like:
 
```python
s = pd.Series([1, None, 2])
s
0    1.0
1    NaN
2    2.0
dtype: float64
s.astype('uint64')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/generic.py", line 5543, in astype
    new_data = self._mgr.astype(dtype=dtype, copy=copy, errors=errors,)
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 595, in astype
    return self.apply("astype", dtype=dtype, copy=copy, errors=errors)
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 406, in apply
    applied = getattr(b, f)(**kwargs)
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/internals/blocks.py", line 594, in astype
    values = astype_nansafe(vals1d, dtype, copy=True)
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/dtypes/cast.py", line 961, in astype_nansafe
    raise ValueError("Cannot convert non-finite values (NA or inf) to integer")
ValueError: Cannot convert non-finite values (NA or inf) to integer
```

But it doesn't in serial execution, fixed the pytest now. Looking at this I think this is a `pytest` issue more than `pytest-xdist` issue. Using `pytest-xdist` is just unearthing the issue with `pytest`. I'll probably come up with a minimal reproducer and raise an issue with pytest.

Reply:
In `test_series_fillna_numerical` were we running an inplace operation against an object that is passed in as a parameter? It would be crazy expensive for pytest to deep copy the object in the parameter for every test invocation so this sounds like expected behavior from my perspective.

Reply:
Yes, we were running an inplace operation on object being passed in as a pytest param